### PR TITLE
Fix malformed headers

### PR DESCRIPTION
**Pull request by steve-chavez**
When having a malformed header such as:

```
$ curl localhost:8080/pathological?malformed-header=header-injection -i
HTTP/1.1 200 OK
Server: nginx/1.26.0

HeaderInjection
Injected-Header: This header contains an injection
```

The background worker crashes.

To prove the above, we use https://github.com/steve-chavez/ngx_pathological.

### CODE REVIEW

**Comment by steve-chavez** [comment]
Tempted to use this newish (2022) curl feature for parsing the headers: https://curl.se/libcurl/c/curl_easy_nextheader.html ([announcement](https://daniel.haxx.se/blog/2022/03/22/a-headers-api-for-libcurl/)).  It's pretty convenient.

It's available from curl 7.83, I believe now that we have Nix on prod it should be doable to get this version.

Note: We need to document this requirement on the README.

**Comment by steve-chavez** [comment]
Some failure about GLIBC on CI https://github.com/supabase/pg_net/actions/runs/10296126418/job/28496960251?pr=136

```
waiting for server to start....2024-08-08 04:53:06.180 UTC [13164] FATAL:  could not load library "/nix/store/hcpahbjn22r4ddbqgsh0pfmgy5vy7lls-postgresql-and-plugins-12.5/lib/pg_net.so": 
/nix/store/1yvpgm763b3hvg8q4fzpzmflr5674x4j-glibc-2.32-10/lib/libc.so.6: version `GLIBC_2.33' not found (required by /nix/store/36kr91f8s509z2r8hqvpayw5bqyqgyzr-curl-8.7.1/lib/libcurl.so.4)
```

Easiest way to solve it is to upgrade to a new nixpkgs version. A bit related to https://github.com/supabase/pg_net/issues/106

**Review by soedirgo — APPROVED** [review]

File: src/worker.c [hunk 1]
```diff
@@ -430,16 +423,14 @@ _PG_init(void)
...
-  worker.bgw_flags = BGWORKER_SHMEM_ACCESS | BGWORKER_BACKEND_DATABASE_CONNECTION;
-  worker.bgw_start_time = BgWorkerStart_RecoveryFinished;
-  snprintf(worker.bgw_library_name, BGW_MAXLEN, "pg_net");
-  snprintf(worker.bgw_function_name, BGW_MAXLEN, "pg_net_worker");
-  snprintf(worker.bgw_name, BGW_MAXLEN, "pg_net " EXTVERSION " worker");
-  worker.bgw_restart_time = 1;
-  worker.bgw_main_arg = (Datum) 0;
-  worker.bgw_notify_pid = 0;
-  RegisterBackgroundWorker(&worker);
+  RegisterBackgroundWorker(&(BackgroundWorker){
```
Comment (pashkinelfe) [inline src/worker.c:426]: IMO previous state of pg_init() was completely ok and also was easier to read.

Comment (steve-chavez) [reply src/worker.c:426]: The new way using C99 designated initializer requires fewer LOC and avoids the needed to `memset` to 0. Readability is highly subjective but fewer LOCs in this case is a win IMO.

**Pull request merged**